src/ibl_mesoscope_to_nwb/mesoscope2025/datainterfaces/_ibl_mesoscope_mc_imaging_extractor.py

```diff
@@ -0,0 +1,240 @@
+from pathlib import Path
+from typing import Optional, Tuple
+
+import numpy as np
+from pydantic import DirectoryPath
+from roiextractors import ImagingExtractor
+
+
+class IBLMesoscopeMotionCorrectedImagingExtractor(ImagingExtractor):
+    """A segmentation extractor for IBL Motion Corrected Mesoscopic imaging data (.bin)."""
+
+    extractor_name = "IBLMesoscopeMotionCorrectedImagingExtractor"
+
+    @classmethod
+    def get_available_planes(cls, folder_path: DirectoryPath) -> list[str]:
+        """Get the available plane names from the folder produced by IBL Mesoscope.
+
+        Parameters
+        ----------
+        folder_path : PathType
+            Path to IBL Mesoscope output path.
+
+        Returns
+        -------
+        plane_names: list
+            List of plane names.
+        """
+        from natsort import natsorted
+
+        folder_path = Path(folder_path)
+        prefix = "plane"
+        plane_paths = natsorted(folder_path.glob(pattern=prefix + "*"))
+        assert len(plane_paths), f"No planes found in '{folder_path}'."
+        plane_names = [plane_path.stem for plane_path in plane_paths]
+        return plane_names
+
+    def __init__(self, file_path: str):
+        """Initialize a IBLMesoscopeMotionCorrectedImagingExtractor instance.
+
+        Main class for extracting imaging data from .bin format.
+
+        Parameters
+        ----------
+        file_path: str or Path
+            Path to the .bin file containing imaging data.
+            Expected path format: .../suite2p/planeX/imaging.frames_motionRegistered.bin
+        """
+        super().__init__(file_path=file_path)
+        self._file_path = Path(file_path)
+        self._channel_names = ["Green"]
```

Comment:
we should align on this, at this moment I'm using `optical_channel_calcium` and `optical_channel_isosbestic` for the name of the optical channels. I'm fine with "Blue" or "Violet" too, but we should be aligned on this. What is your preference? Can you look at [my metadata structure](https://github.com/catalystneuro/IBL-widefield-to-nwb/blob/5b4c21b87dc53d0e6751c177ef82cd971b03180d/src/ibl_widefield_to_nwb/widefield2025/metadata/widefield_ophys_metadata.yaml) for reference?